readthedocs/subscriptions/event_handlers.py

```diff
@@ -0,0 +1,120 @@
+"""
+Dj-stripe webhook handlers.
+
+https://docs.dj-stripe.dev/en/master/usage/webhooks/.
+"""
+import structlog
+from django.conf import settings
+from django.utils import timezone
+from djstripe import models as djstripe
+from djstripe import webhooks
+from djstripe.enums import SubscriptionStatus
+
+from readthedocs.organizations.models import Organization
+from readthedocs.payments.utils import cancel_subscription as cancel_stripe_subscription
+from readthedocs.subscriptions.models import Subscription
+
+log = structlog.get_logger(__name__)
+
+
+def _update_subscription_from_stripe(rtd_subscription, stripe_subscription_id):
+    """Update the RTD subscription object given the new stripe subscription object."""
+    log.bind(stripe_subscription_id=stripe_subscription_id)
+    stripe_subscription = djstripe.Subscription.objects.filter(
+        id=stripe_subscription_id
+    ).first()
+    if not stripe_subscription:
+        log.info("Stripe subscription not found.")
+        return
+
+    previous_subscription_id = rtd_subscription.stripe_id
+    Subscription.objects.update_from_stripe(
+        rtd_subscription=rtd_subscription,
+        stripe_subscription=stripe_subscription,
+    )
+    log.info(
+        "Subscription updated.",
+        previous_stripe_subscription_id=previous_subscription_id,
+        stripe_subscription_status=stripe_subscription.status,
+    )
+
+    # Cancel the trial subscription if its trial has ended.
+    trial_ended = (
+        stripe_subscription.trial_end and stripe_subscription.trial_end < timezone.now()
+    )
+    is_trial_subscription = stripe_subscription.items.filter(
+        price__id=settings.RTD_ORG_DEFAULT_STRIPE_SUBSCRIPTION_PRICE
+    ).exists()
+    if (
+        is_trial_subscription
+        and trial_ended
+        and stripe_subscription.status != SubscriptionStatus.canceled
+    ):
+        log.info(
+            "Trial ended, canceling subscription.",
+        )
+        cancel_stripe_subscription(stripe_subscription.id)
+
+
+@webhooks.handler("customer.subscription.updated", "customer.subscription.deleted")
+def update_subscription(event):
+    """Update the RTD subscription object given the new stripe subscription."""
```

Comment:
```suggestion
    """Update the RTD subscription object with the updates from the Stripe subscription."""
```

Just to mention that it's not a _new_ Stripe subscription, but it's just updated.